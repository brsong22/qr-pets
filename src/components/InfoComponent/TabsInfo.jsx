import React, { Component } from 'react';
import PropTypes from 'prop-types';
import InfoTab from './InfoTab';

class TabsInfo extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { tabs, info, onTab } = this.props;
    const i = tabs.map((tab, x) => (
      <InfoTab key={`infoText_${tab}`} tabText={tab} infoText={info[x]} onTab={onTab} />
    ));
    return (
      <div className="infoTab">
        {i}
      </div>
    );
  }
}

TabsInfo.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
  info: PropTypes.arrayOf(PropTypes.string),
  onTab: PropTypes.string,
};

TabsInfo.defaultProps = {
  info: [],
  onTab: 'info',
};

export default TabsInfo;
