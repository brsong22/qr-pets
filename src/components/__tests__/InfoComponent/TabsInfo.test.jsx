
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TabsInfo from '../../InfoComponent/TabsInfo';

describe('TabsInfo', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<TabsInfo tabs={['test1', 'test2', 'test3']} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
