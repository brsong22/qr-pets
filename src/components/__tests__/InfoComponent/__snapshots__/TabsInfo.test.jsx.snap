// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsInfo renders as expected 1`] = `
<div
  className="infoTab"
>
  <InfoTab
    infoText=""
    key="infoText_test1"
    onTab="info"
    tabText="test1"
  />
  <InfoTab
    infoText=""
    key="infoText_test2"
    onTab="info"
    tabText="test2"
  />
  <InfoTab
    infoText=""
    key="infoText_test3"
    onTab="info"
    tabText="test3"
  />
</div>
`;
